### PR TITLE
add controllers, dto, mappers for Movie, CinemaHall, MovieSession

### PR DESCRIPTION
**Pull request by Alkey**

### CODE REVIEW

File: src/main/java/com/dev/cinema/controllers/AuthenticationController.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+        this.service = authenticationService;
+    }
+
+    @PostMapping("/registration")
+    public void registration(@RequestBody UserRequestDto dto) {
+        try {
+            service.register(dto.getEmail(), dto.getPassword());
+        } catch (AuthenticationException e) {
+            logger.error("Can't register user " + dto, e);
+        }
```
Comment (dariiabutorina) [inline src/main/java/com/dev/cinema/controllers/AuthenticationController.java]: Isn't it better to log the exception in the dao or at least service layer?
Definitely not in the controller

File: src/main/java/com/dev/cinema/mapper/MovieMapper.java [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+import com.dev.cinema.model.Movie;
+import com.dev.cinema.model.dto.MovieRequestDto;
+import com.dev.cinema.model.dto.MovieResponseDto;
+import org.springframework.stereotype.Component;
+
+@Component
+public class MovieMapper {
+
+    public Movie getMovieFromRequestDto(MovieRequestDto dto) {
```
Comment (dariiabutorina) [inline src/main/java/com/dev/cinema/mapper/MovieMapper.java]:
```suggestion
public class MovieMapper {
    public Movie getMovieFromRequestDto(MovieRequestDto dto) {
```

File: src/main/java/com/dev/cinema/mapper/UserMapper.java [hunk 3]
```diff
@@ -0,0 +1,15 @@
+package com.dev.cinema.mapper;
+
+import com.dev.cinema.model.User;
+import com.dev.cinema.model.dto.UserRequestDto;
+import org.springframework.stereotype.Component;
+
+@Component
+public class UserMapper {
```
Comment (RomanDubovskyi) [inline src/main/java/com/dev/cinema/mapper/UserMapper.java]: we don't need this mapper for now, let's remove it

File: src/main/java/com/dev/cinema/controllers/AuthenticationController.java [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+    @Autowired
+    public AuthenticationController(AuthenticationService authenticationService) {
+        this.service = authenticationService;
+    }
+
+    @PostMapping("/registration")
+    public void registration(@RequestBody UserRequestDto dto) {
+        try {
+            service.register(dto.getEmail(), dto.getPassword());
```
Comment (RomanDubovskyi) [inline src/main/java/com/dev/cinema/controllers/AuthenticationController.java]: why do we have Authentication exception while registration?

**Review by RomanDubovskyi — APPROVED** [review]

**Pull request merged**